tests/functional/test_vcs_git.py

```diff
@@ -282,3 +283,97 @@ def test_resolve_commit_not_on_branch(script, tmp_path):
     # check we can fetch our commit
     rev_options = Git.make_rev_options(commit)
     Git().fetch_new(str(clone_path), repo_path.as_uri(), rev_options)
+
+
+def _initialize_clonetest_server(repo_path, script):
+    repo_path.mkdir()
+    script.run("git", "init", cwd=str(repo_path))
+    repo_file = repo_path / "file.txt"
+    repo_file.write_text(u".")
+    script.run("git", "add", "file.txt", cwd=str(repo_path))
+    script.run("git", "commit", "-m", "initial commit", cwd=str(repo_path))
+    return repo_file
+
+
+@pytest.mark.skipif(Git().get_git_version() < (2, 17), reason="git too old")
+def test_partial_clone(script, tmp_path):
+    """Test partial clone w/ a git-server that supports it"""
+    repo_path = tmp_path / "repo"
+    repo_file = _initialize_clonetest_server(repo_path, script)
+    clone_path1 = repo_path / "clone1"
+    clone_path2 = repo_path / "clone2"
+
+    commit = script.run("git", "rev-parse", "HEAD", cwd=str(repo_path)).stdout.strip()
+
+    # Enable filtering support on server
+    script.run("git", "config", "uploadpack.allowFilter", "true", cwd=repo_path)
+    script.run("git", "config", "uploadpack.allowanysha1inwant", "true", cwd=repo_path)
```

Comment:
Better to put this in the server initialisation function instead (either toggling with a flag e.g. `setup_filter=True`) or a new function e.g. `_initialize_clonetest_server_with_filter`